src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -654,14 +654,18 @@ private function resolveServices($value, $file, $isParameter = false)
             $argument = $value->getValue();
             if ('iterator' === $value->getTag()) {
                 if (!is_array($argument)) {
-                    throw new InvalidArgumentException('"!iterator" tag only accepts sequences.');
+                    throw new InvalidArgumentException(sprintf('"!iterator" tag only accepts sequences in "%s".', $file));
                 }
 
-                return new IteratorArgument($this->resolveServices($argument, $file, $isParameter));
+                try {
+                    return new IteratorArgument($this->resolveServices($argument, $file, $isParameter));
```

Comment:
IMO, `resolveServices` should be outside the try/catch, so that we don't replace its exceptions by mistake